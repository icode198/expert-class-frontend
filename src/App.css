.logOutBtn {
  position: absolute;
  bottom: 0;
  margin: 20px;
  height: 2rem;
  width: 5rem;
  font-size: 0.8rem;
  border-radius: 50px;
  border: 2px solid white;
  font-weight: 500;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  background-color: white;
  color: rgba(151, 191, 15, 1);
  cursor: pointer;
}

.logOutBtn:hover {
  color: white;
  background-color: rgba(151, 191, 15, 0.7);
}
