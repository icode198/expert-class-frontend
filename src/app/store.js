import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import classesReducer from '../features/classes/classesSlice';
import reservationsReducer from '../features/reservations/reservationsSlice';
import sessionReducer from '../auth/sessionSlice';
import reservePageReducer from '../features/reserve/reservePageSlice';

const store = configureStore({
  reducer: {
    classes: classesReducer,
    reservations: reservationsReducer,
    users: sessionReducer,
    reserve: reservePageReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
