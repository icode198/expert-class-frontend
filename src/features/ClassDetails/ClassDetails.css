.details-container {
  display: flex;
  width: 1050px;
  height: 100vh;
  margin-left: 250px;
}

.details-col-1 {
  width: 75%;
  padding: 20px;
}

.details-col-1 img {
  width: 100%;
  height: 100%;
}

.details-col-2 {
  width: 25%;
  margin: 0 10px;
}

.details-col-2 .title {
  text-align: right;
  padding: 15px;
}

.details-col-2 h4 {
  padding: 15px;
  border: 1px solid #eee;
}

.details-col-2 h4:nth-child(even) {
  background-color: #eee;
}
