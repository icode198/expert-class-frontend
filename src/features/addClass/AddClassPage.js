import React, { useState } from 'react';
import axios from 'axios';
import getCSRFToken from '../../app/getCSRFToken';
import expertClassAPI from '../../app/expertClassAPI';

const AddClassPage = () => {
  const handleSubmit = async (event) => {
    event.preventDefault();
    const formData = new FormData(event.target);

    await axios.post('http://localhost:3001/api/v1/courses',
      formData,
      {
        withCredentials: true,
        headers: {
          'X-CSRF-Token': getCSRFToken(),
          'content-type': 'multipart/form-data',
        },
      });
  };

  const [classes, setClasses] = useState(null);

  const [course, setCourse] = useState(null);

  const classesList = (courses) => courses.map((c) => (
    <div key={c.id}>
      <img src={c.course_image_url} alt="class instructor" />
      <h3>{c.title}</h3>
      <p>{c.description}</p>
    </div>
  ));

  const showClass = (c) => (
    <div key={c.id}>
      <img src={c.course_image_url} alt="class instructor" />
      <h3>{c.title}</h3>
      <p>{c.description}</p>
    </div>
  );

  const fetchClasses = async () => {
    const response = await expertClassAPI.get('/api/v1/courses');
    setClasses(classesList(response.data));
  };

  const fetchClass = async () => {
    const response = await expertClassAPI.get('/api/v1/courses/16');
    setCourse(showClass(response.data));
  };

  return (
    <>
      <div style={{ marginLeft: 400 }}>
        <h1>Create a New Class</h1>
        <form onSubmit={handleSubmit}>
          <label htmlFor="title">
            Title
            <input type="text" name="course[title]" />
          </label>
          <label htmlFor="description">
            Description
            <input type="text" name="course[description]" />
          </label>
          <label htmlFor="duration">
            Duration
            <input type="number" name="course[duration]" />
          </label>
          <label htmlFor="instructor">
            Instructor
            <input type="text" name="course[instructor]" />
          </label>
          <label htmlFor="image">
            Upload image of Instructor
            <input type="file" name="course[image]" accept="image/png, image/jpeg" />
          </label>
          <input type="submit" value="Upload" />
        </form>
      </div>

      <div style={{
        width: 600, marginLeft: 400, marginTop: 100, minHeight: 200,
      }}
      >
        <button type="button" onClick={fetchClasses}>Fetch Classes</button>
        {classes}
      </div>
      <div style={{
        width: 600, marginLeft: 400, marginTop: 100, minHeight: 200,
      }}
      >
        <button type="button" onClick={fetchClass}>Fetch One Class</button>
        {course}
      </div>
    </>
  );
};

export default AddClassPage;
