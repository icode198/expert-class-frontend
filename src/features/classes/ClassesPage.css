* {
  text-decoration: none;
}

.classes-container {
  width: 1070px;
  height: 100vh;
  padding: 20px;
  margin-left: 240px;
}

.heading {
  text-align: center;
  margin: 25px auto;
  text-transform: uppercase;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.carousel-container {
  height: 100%;
}

.carousel-component {
  height: 100%;
}

.classes-item {
  width: 300px;
  border-radius: 15px;
  background-color: #97bf0f;
  color: #fff;
}

.text-container,
.image-container {
  margin: 0 auto;
  text-align: center;
}

.text-container {
  padding: 20px;
}

.image {
  margin: 0 auto;
  width: 100%;
  height: 300px;
}

.classes-title {
  text-transform: uppercase;
}

.class-instructor {
  padding: 0.5rem;
}

.classes-social-btns {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 1rem;
}

.classes-social-icon-circle {
  width: 26px;
  height: 26px;
  padding: 4px;
  margin: 0 5px;
  border: 2px solid lightgreen;
  border-radius: 50%;
  cursor: pointer;
}

.classes-social-icon {
  color: lightgreen;
  fill: lightgreen;
}
