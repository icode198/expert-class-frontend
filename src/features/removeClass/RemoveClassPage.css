.remove-class-container {
  width: 1050px;
  height: 100vh;
  margin-left: 250px;
  padding: 10px;
}

th,
td {
  padding: 10px;
  text-align: left;
  border: 1px solid #333;
}

th {
  background-color: #97bf0f;
}

table tr:nth-child(odd) {
  background-color: #eee;
}