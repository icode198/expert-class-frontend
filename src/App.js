import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { HashRouter as Router, Switch, Route } from 'react-router-dom';
import axios from 'axios';
import { loginStatus, loginUser, logoutUser, currentUser, loggedInStatus } from './auth/sessionSlice';
import { isCSRFToken } from './app/getCSRFToken';
import ClassesPage from './features/classes/ClassesPage'; // eslint-disable-line
import LandingPage from './features/landingPage/landingPage';
import Session from './auth/session';
import NavPanel from './features/navPanel/NavPanel';
import ReservePage from './features/reserve/ReservePage';
import ReservationsPage from './features/reservations/ReservationsPage'; // eslint-disable-line
import AddClassPage from './features/addClass/AddClassPage';
import RemoveClassPage from './features/removeClass/RemoveClassPage';

function App() {
  console.log('isCSRFToken?: ', (isCSRFToken()) ? 'true' : 'false');
  const dispatch = useDispatch();

  const user = useSelector(currentUser);
  const loggedIn = useSelector(loggedInStatus)

  return (
    <div className="App">
      <Router>
        <NavPanel />
      </Router>
      <div className="app-content">
        <Router>
          <Switch>
            <Route path="/classes">
              <ClassesPage />
            </Route>
            <Route path="/reserve">
              <ReservePage />
            </Route>
            <Route path="/my-reservations">
              <ReservationsPage />
            </Route>
            <Route path="/add-class">
              <AddClassPage />
            </Route>
            <Route path="/remove-class">
              <RemoveClassPage />
            </Route>
            <Route path="/">
              {/* <LandingPage /> */}
              <Session />
            </Route>
          </Switch>
        </Router>
      </div>
    </div>
  );
}

export default App;
