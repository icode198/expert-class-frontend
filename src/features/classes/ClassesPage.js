import './ClassesPage.css';
import '@brainhubeu/react-carousel/lib/style.css';
import Carousel, { slidesToShowPlugin, autoplayPlugin } from '@brainhubeu/react-carousel';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchClassesData } from './classesSlice';
import { reloadClasses } from '../removeClass/removeClassSlice';
import SocialIcons from './socialIcons';

export default function ClassesPage() {
  const dispatch = useDispatch();

  const classes = useSelector((state) => state.classes.classes);

  useEffect(() => {
    dispatch(fetchClassesData());
    dispatch(reloadClasses());
  }, []);

  const classesList = classes.map((c) => (
    <NavLink to={`/class/${c.id}`} key={c.id} className="classes-item">
      <div className="image-container">
        <img className="image" src={c.course_image_url} alt="course" />
      </div>
      <div className="text-container">
        <h3 className="classes-title">{c.title}</h3>
        <p className="class-instructor">{c.instructor}</p>
        <SocialIcons />
      </div>
    </NavLink>
  ));

  return (
    <div className="classes-container">
      <h2 className="heading">Classes</h2>
      <Carousel
        plugins={[
          'centered',
          'infinite',
          'arrows',
          {
            resolve: slidesToShowPlugin,
            options: {
              numberOfSlides: 3,
            },
          },
          {
            resolve: autoplayPlugin,
            options: {
              interval: 3000,
            },
          },
        ]}
        animationSpeed={1000}
        className="carousel-component"
      >
        {classesList}
      </Carousel>
    </div>
  );
}
