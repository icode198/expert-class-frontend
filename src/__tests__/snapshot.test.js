import React from 'react';
import TestRenderer from 'react-test-renderer';
import { act, render } from '@testing-library/react';
import { Provider, useSelector } from 'react-redux';
import ReactRouter from 'react-router';
import {
  HashRouter as Router, Switch, Route,
} from 'react-router-dom';
import App from '../App';
import store from '../app/store';
import LandingPage from '../pages/landingPage/landingPage';
import SignInPage from '../pages/signInPage/signInPage';
import SignUpPage from '../pages/signUpPage/signUpPage';
import ClassesPage from '../pages/classesPage/classesPage';
import ClassDetails from '../pages/classDetailsPage/classDetailsPage';
import ReservePage from '../pages/reservePage/reservePage';
import ReservationsPage from '../pages/reservationsPage/reservationsPage';
import AddClassPage from '../pages/addClassPage/addClassPage';
import RemoveClassPage from '../pages/removeClassPage/removeClassPage';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('App', () => {
  test('Renders the App', () => {
    const app = TestRenderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});

describe('LandingPage', () => {
  test('Renders the page', () => {
    const app = TestRenderer.create(
      <Provider store={store}>
        <LandingPage />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});

describe('SignIn Page', () => {
  test('Renders the page', () => {
    jest.spyOn(ReactRouter, 'useLocation').mockReturnValue({ state: null });

    const app = TestRenderer.create(
      <Provider store={store}>
        <SignInPage />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});

describe('SignUp Page', () => {
  test('Renders the page', () => {
    const app = TestRenderer.create(
      <Provider store={store}>
        <SignUpPage />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});

describe('Classes Page', () => {
  beforeEach(() => {
    const mockAppState = {
      classes: {
        classes: [
          {
            course_image_url: 'https://res.cloudinary.com/hhh6cnm2e/image/upload/11ijaidwrn9s7s5gogab9j6wsmnz.jpeg',

            created_at: '2021-11-11T04:20:38.165Z',

            description: 'Learn how to draft professional essays',

            duration: 4,

            id: 19,

            instructor: 'Jeremy Campbell',

            title: 'English Composition',

            updated_at: '2021-11-11T04:20:38.181Z',
          },
        ],
        status: 'fulfilled',
        error: null,
      },

      users: {
        user: {
          id: 39,
          username: 'arturo_coder',
          name: 'Arturo',
          created_at: '2021-11-05T23:43:07.938Z',
          updated_at: '2021-11-05T23:43:07.938Z',
        },
        logged_in: true,
        status: 'fulfilled',
        error: null,
      },
    };

    useSelector.mockImplementation((callback) => callback(mockAppState));

    TestRenderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
  });

  test('Renders the page', () => {
    const page = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <ClassesPage />
        </Router>
      </Provider>,
    ).toJSON();

    expect(page).toMatchSnapshot();
  });
});

describe('Class Details Page', () => {
  beforeEach(() => {
    const mockAppState = {
      classes: {
        classes: [
          {
            course_image_url: 'https://res.cloudinary.com/hhh6cnm2e/image/upload/11ijaidwrn9s7s5gogab9j6wsmnz.jpeg',

            created_at: '2021-11-11T04:20:38.165Z',

            description: 'Learn how to draft professional essays',

            duration: 4,

            id: 19,

            instructor: 'Jeremy Campbell',

            title: 'English Composition',

            updated_at: '2021-11-11T04:20:38.181Z',
          },
        ],
        status: 'fulfilled',
        error: null,
      },
      classDetails: {
        classObj: [],
        error: null,
        status: 'idle',
      },
      users: {
        user: {
          id: 39,
          username: 'arturo_coder',
          name: 'Arturo',
          created_at: '2021-11-05T23:43:07.938Z',
          updated_at: '2021-11-05T23:43:07.938Z',
        },
        logged_in: true,
        status: 'fulfilled',
        error: null,
      },
    };

    useSelector.mockImplementation((callback) => callback(mockAppState));

    TestRenderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
  });

  test('Renders the page', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ id: 19 });

    const page = TestRenderer.create(
      <Provider store={store}>
        <ClassDetails />
      </Provider>,
    ).toJSON();
    expect(page).toMatchSnapshot();
  });
});

// describe('Reserve Page', () => {
//   test('Renders the page', () => {
//     const page = TestRenderer.create(
//       <Provider store={store}>
//         <ReservePage />
//       </Provider>,
//     ).toJSON();
//     expect(page).toMatchSnapshot();
//   });
// });

// describe('Reservations Page', () => {
//   test('Renders the page', () => {
//     const page = TestRenderer.create(
//       <Provider store={store}>
//         <ReservationsPage />
//       </Provider>,
//     ).toJSON();
//     expect(page).toMatchSnapshot();
//   });
// });

// describe('Add Class Page', () => {
//   test('Renders the page', () => {
//     const page = TestRenderer.create(
//       <Provider store={store}>
//         <AddClassPage />
//       </Provider>,
//     ).toJSON();
//     expect(page).toMatchSnapshot();
//   });
// });

// describe('Remove Class Page', () => {
//   test('Renders the page', () => {
//     const page = TestRenderer.create(
//       <Provider store={store}>
//         <RemoveClassPage />
//       </Provider>,
//     ).toJSON();
//     expect(page).toMatchSnapshot();
//   });
// });
