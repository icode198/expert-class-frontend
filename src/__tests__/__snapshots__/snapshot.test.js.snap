// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders the App 1`] = `null`;

exports[`Class Details Page Renders the page 1`] = `
<div
  className="details-container"
>
  <div
    className="details-col-1"
  >
    <img
      alt="expert class"
    />
  </div>
  <div
    className="details-col-2"
  >
    <div
      className="title"
    >
      <h2
        className="class-title"
      />
      <h3
        className="class-instructor"
      />
    </div>
    <div
      className="desc-table"
    >
      <h4>
        <svg
          className="bi bi-alarm"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 5.5a.5.5 0 0 0-1 0v3.362l-1.429 2.38a.5.5 0 1 0 .858.515l1.5-2.5A.5.5 0 0 0 8.5 9V5.5z"
          />
          <path
            d="M6.5 0a.5.5 0 0 0 0 1H7v1.07a7.001 7.001 0 0 0-3.273 12.474l-.602.602a.5.5 0 0 0 .707.708l.746-.746A6.97 6.97 0 0 0 8 16a6.97 6.97 0 0 0 3.422-.892l.746.746a.5.5 0 0 0 .707-.708l-.601-.602A7.001 7.001 0 0 0 9 2.07V1h.5a.5.5 0 0 0 0-1h-3zm1.038 3.018a6.093 6.093 0 0 1 .924 0 6 6 0 1 1-.924 0zM0 3.5c0 .753.333 1.429.86 1.887A8.035 8.035 0 0 1 4.387 1.86 2.5 2.5 0 0 0 0 3.5zM13.5 1c-.753 0-1.429.333-1.887.86a8.035 8.035 0 0 1 3.527 3.527A2.5 2.5 0 0 0 13.5 1z"
          />
        </svg>
        <span>
          Hour(s)
        </span>
      </h4>
      <h4>
        <svg
          className="bi bi-megaphone"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 2.5a1.5 1.5 0 0 1 3 0v11a1.5 1.5 0 0 1-3 0v-.214c-2.162-1.241-4.49-1.843-6.912-2.083l.405 2.712A1 1 0 0 1 5.51 15.1h-.548a1 1 0 0 1-.916-.599l-1.85-3.49a68.14 68.14 0 0 0-.202-.003A2.014 2.014 0 0 1 0 9V7a2.02 2.02 0 0 1 1.992-2.013 74.663 74.663 0 0 0 2.483-.075c3.043-.154 6.148-.849 8.525-2.199V2.5zm1 0v11a.5.5 0 0 0 1 0v-11a.5.5 0 0 0-1 0zm-1 1.35c-2.344 1.205-5.209 1.842-8 2.033v4.233c.18.01.359.022.537.036 2.568.189 5.093.744 7.463 1.993V3.85zm-9 6.215v-4.13a95.09 95.09 0 0 1-1.992.052A1.02 1.02 0 0 0 1 7v2c0 .55.448 1.002 1.006 1.009A60.49 60.49 0 0 1 4 10.065zm-.657.975 1.609 3.037.01.024h.548l-.002-.014-.443-2.966a68.019 68.019 0 0 0-1.722-.082z"
          />
        </svg>
        <span />
      </h4>
    </div>
  </div>
</div>
`;

exports[`Classes Page Renders the page 1`] = `
<div
  className="classesContainer"
>
  <h2
    className="heading"
  >
    Classes
  </h2>
  <div
    className="swiper"
    id="swiper-container"
  >
    <div
      className="swiper-wrapper"
    >
      <div
        className="swiper-slide swiperSlide swiperSlideSm"
      >
        <a
          aria-current={null}
          className="swiperCard"
          href="#/class/19"
          onClick={[Function]}
        >
          <div
            className="swiperImgContainer"
          >
            <img
              alt="course instructor"
              className="swiperImg"
              src="https://res.cloudinary.com/hhh6cnm2e/image/upload/11ijaidwrn9s7s5gogab9j6wsmnz.jpeg"
            />
          </div>
          <div
            className="textContainer"
          >
            <h3
              className="title"
            >
              English Composition
            </h3>
            <p
              className="instructor"
            >
              Jeremy Campbell
            </p>
            <div
              className="socialBtns"
            >
              <span
                className="socialIconCircle"
              >
                <svg
                  className="socialIcon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Facebook
                  </title>
                  <path
                    d="M24 12.073c0-6.627-5.373-12-12-12s-12 5.373-12 12c0 5.99 4.388 10.954 10.125 11.854v-8.385H7.078v-3.47h3.047V9.43c0-3.007 1.792-4.669 4.533-4.669 1.312 0 2.686.235 2.686.235v2.953H15.83c-1.491 0-1.956.925-1.956 1.874v2.25h3.328l-.532 3.47h-2.796v8.385C19.612 23.027 24 18.062 24 12.073z"
                  />
                </svg>
              </span>
              <span
                className="socialIconCircle"
              >
                <svg
                  className="socialIcon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Twitter
                  </title>
                  <path
                    d="M23.953 4.57a10 10 0 01-2.825.775 4.958 4.958 0 002.163-2.723c-.951.555-2.005.959-3.127 1.184a4.92 4.92 0 00-8.384 4.482C7.69 8.095 4.067 6.13 1.64 3.162a4.822 4.822 0 00-.666 2.475c0 1.71.87 3.213 2.188 4.096a4.904 4.904 0 01-2.228-.616v.06a4.923 4.923 0 003.946 4.827 4.996 4.996 0 01-2.212.085 4.936 4.936 0 004.604 3.417 9.867 9.867 0 01-6.102 2.105c-.39 0-.779-.023-1.17-.067a13.995 13.995 0 007.557 2.209c9.053 0 13.998-7.496 13.998-13.985 0-.21 0-.42-.015-.63A9.935 9.935 0 0024 4.59z"
                  />
                </svg>
              </span>
              <span
                className="socialIconCircle"
              >
                <svg
                  className="socialIcon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Instagram
                  </title>
                  <path
                    d="M12 0C8.74 0 8.333.015 7.053.072 5.775.132 4.905.333 4.14.63c-.789.306-1.459.717-2.126 1.384S.935 3.35.63 4.14C.333 4.905.131 5.775.072 7.053.012 8.333 0 8.74 0 12s.015 3.667.072 4.947c.06 1.277.261 2.148.558 2.913.306.788.717 1.459 1.384 2.126.667.666 1.336 1.079 2.126 1.384.766.296 1.636.499 2.913.558C8.333 23.988 8.74 24 12 24s3.667-.015 4.947-.072c1.277-.06 2.148-.262 2.913-.558.788-.306 1.459-.718 2.126-1.384.666-.667 1.079-1.335 1.384-2.126.296-.765.499-1.636.558-2.913.06-1.28.072-1.687.072-4.947s-.015-3.667-.072-4.947c-.06-1.277-.262-2.149-.558-2.913-.306-.789-.718-1.459-1.384-2.126C21.319 1.347 20.651.935 19.86.63c-.765-.297-1.636-.499-2.913-.558C15.667.012 15.26 0 12 0zm0 2.16c3.203 0 3.585.016 4.85.071 1.17.055 1.805.249 2.227.415.562.217.96.477 1.382.896.419.42.679.819.896 1.381.164.422.36 1.057.413 2.227.057 1.266.07 1.646.07 4.85s-.015 3.585-.074 4.85c-.061 1.17-.256 1.805-.421 2.227-.224.562-.479.96-.899 1.382-.419.419-.824.679-1.38.896-.42.164-1.065.36-2.235.413-1.274.057-1.649.07-4.859.07-3.211 0-3.586-.015-4.859-.074-1.171-.061-1.816-.256-2.236-.421-.569-.224-.96-.479-1.379-.899-.421-.419-.69-.824-.9-1.38-.165-.42-.359-1.065-.42-2.235-.045-1.26-.061-1.649-.061-4.844 0-3.196.016-3.586.061-4.861.061-1.17.255-1.814.42-2.234.21-.57.479-.96.9-1.381.419-.419.81-.689 1.379-.898.42-.166 1.051-.361 2.221-.421 1.275-.045 1.65-.06 4.859-.06l.045.03zm0 3.678c-3.405 0-6.162 2.76-6.162 6.162 0 3.405 2.76 6.162 6.162 6.162 3.405 0 6.162-2.76 6.162-6.162 0-3.405-2.76-6.162-6.162-6.162zM12 16c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4zm7.846-10.405c0 .795-.646 1.44-1.44 1.44-.795 0-1.44-.646-1.44-1.44 0-.794.646-1.439 1.44-1.439.793-.001 1.44.645 1.44 1.439z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`LandingPage Renders the page 1`] = `
<div
  className="mainContainer"
>
  <div
    className="innerContainer"
  >
    <h1
      className="pageTitle"
    >
      OUR EXPERT CLASSES
    </h1>
    <p
      className="caption"
    >
      There's Nothing Better Than Getting Better. Keep Growing With Unlimited Creative Classes. Grow Without Limits. Discover What You Can Create With Thousands of Inspiring Classes. Support of a Community. Prompts & Resources. Browse Projects. Offline viewing.
    </p>
    <div
      className="buttonsDiv"
    >
      <button
        className="button1"
        onClick={[Function]}
        type="button"
      >
        SignIn
      </button>
      <button
        className="button2"
        onClick={[Function]}
        type="button"
      >
        SignUp
      </button>
    </div>
  </div>
</div>
`;

exports[`SignIn Page Renders the page 1`] = `
<div
  className="mainContainer"
>
  <div
    className="innerContainer"
  >
    <h2
      className="title"
    >
      Sign In
    </h2>
    <form
      className="signInForm"
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        name="username"
        onChange={[Function]}
        placeholder="Please enter your username"
        required={true}
        type="text"
        value=""
      />
      <input
        className="formSubmit"
        type="submit"
        value="Sign in"
      />
    </form>
  </div>
</div>
`;

exports[`SignUp Page Renders the page 1`] = `
<div
  className="mainContainer"
>
  <div
    className="innerContainer"
  >
    <h2
      className="title"
    >
      Sign Up
    </h2>
    <form
      className="signInForm"
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        id="username"
        maxLength="20"
        minLength="6"
        name="username"
        onChange={[Function]}
        placeholder="Please enter your username"
        required={true}
        type="text"
        value=""
      />
      <input
        className="nameInput"
        id="name"
        maxLength="24"
        name="name"
        onChange={[Function]}
        placeholder="Please enter your name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="formSubmit"
        type="submit"
        value="Sign up"
      />
    </form>
  </div>
</div>
`;
